### DISCUSSION TITLE
stats, performance, database, cache

### DISCUSSION THREAD

**Discussion by adamus007p** [PrestaShop/PrestaShop#37632] [Ideas]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I would like to give idea to improve stats and connections in Prestashop. Now when you even take global stats from data mining module you can have milion entries in the table. Which make a database bigger and bigger. I assume that for Prestashop you want to have fast ecommerce in the market. When you turn off the all stats you do not see visits. Why there is no cache that it would copy numbers from past months years and clean the table. If you want to see stats you would get it fast. I assume that it would be some updates, esspecialy 6months back or up to 12months, but later years you will not change. Is it possible to keep it just a table with stats? and delete old stats? There should be such option and cron to do it.

2. Other thing is keeping in sources the last sources, so usually it is a Paypal or Stipe with some URL. It is make no sense to keep it. More important is to keep the 1st source. There should be an option to choose it.

That two things is would to save a lot of space in the database and keep prestashop faster.

In presta for each quest presta make entry in database but when guest is looged it do not clean it. This also case a lot of entries. Guest should be tracket and converted into customer, now it is not working

### Steps to reproduce

like above

### PrestaShop version(s) where the bug happened

all prestashop

**Comment by paulnoelcholot** [comment]
Hello,

Thank you for your suggestion. The Maintainer Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you create it. 

If you or anyone else is willing to [contribute this feature](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/) to the PrestaShop Project, the teams will be happy to help by ensuring that it fits well within the project. Please let us know if that is the case!

Thank you
